helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/InstancesAccessor.java

```diff
@@ -235,7 +239,22 @@ private Response batchGetStoppableInstances(String clusterId, JsonNode node, boo
             OBJECT_MAPPER.getTypeFactory().constructCollectionType(List.class, String.class));
         if (!orderOfZone.isEmpty() && random) {
```

Comment:
This will break the backward compatibility. User can have non zone selected by let us to choose in order first one.